content/learn/book/ecs/resources/_index.md

```diff
@@ -6,12 +6,296 @@ page_template = "book-section.html"
 
-TODO: Explain why you might want a resource
+You might want to use resources for:
 
-TODO: Show how to create and access Resources
+- storing events
+- reading input data
+- recording game configuration (such as the current difficulty)
+- storing simple global game state that you only need a single copy of (like the player's current score)
+- interoperating with other non-Bevy Rust libraries
+- storing data structures that help you look up entities by their component values more quickly (such as indexes or graphs)
 
-TODO: Show how to modify resources
+[`Res`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.Res.html
+[`ResMut`]: https://docs.rs/bevy/latest/bevy/ecs/system/struct.ResMut.html
 
-TODO: Compare and contrast with `query.single()`
+## Creating resources
+
+Unlike components, resources do not need their own trait implementation: you can use any new or existing [`'static`] Rust type as a resource (if it is not [`Send + Sync`], you'll need a [`NonSend`] resource instead).
+
+Like entities and their component data, resources are stored in your [`App`]'s [`World`] struct.
+Resources are typically added statically, via [`insert_resouce`] or [`init_resource`].
+[`insert_resource`] is used when you want to set the value of a resource manually, while [`init_resource`] is used when you want to automatically initialize the resource's value using the [`Default`] or [`FromWorld`] trait.
+
+```rust
+use bevy::prelude::*;
+
+// Default can be derived for many simple resources,
+// with the default value of most numeric types being 0
+#[derive(Default)]
+struct Score(u64);
+
+struct PlayerSupplies {
+    gold: u64,
+    wood: u64,
+}
+
+// The Default trait can be manually implemented to control initial values
+impl Default for PlayerSupplies {
+    fn default() -> Self {
+        PlayerSupplies {
+            gold: 400,
+            wood: 200,
+        }
+    }
+}
+
+// Enum resources are a great way to represent game state in a type-safe way
+enum Turn {
+    Allied,
+    Enemy,
+}
+
+fn main() {
+    // Resources are typically inserted using methods on `App`
+    App::build()
+        // Uses the default() value provided by the derived Default trait
+        .init_resource::<Score>()
+        // Uses the default() value provided by the manual impl of the Default trait
+        .init_resource::<PlayerSupplies>()
+        // Uses the specific value supplied (Turn::Allied) to insert a resource of type Turn
+        .insert_resource(Turn::Allied)
+        // Sets the value of the standard Bevy resource `WindowDescriptor`,
+        // leaving the unspecified fields as their default value
+        .insert_resource(WindowDescriptor {
+            title: "I am a window!".to_string(),
+            width: 500.,
+            height: 300.,
+            vsync: true,
+            ..default()
+        })
+        .add_plugins(DefaultPlugins)
+        .run()
+}
+```
+
+In rare cases, you may need to add a resource after app startup.
```

Comment:
I think I would extract that entire paragraph to it's own section, at least to make it easier to link to it.